### ISSUE TITLE
HTTP 400 error when downloading specific podcast files

### ISSUE THREAD

**Issue by usmanatron** [gpodder/gpodder#1296]
Hello,

I've got a few podcast files which continuously fail to download, even after retrying etc.  Looking at the logs, I've got this output:

```
2022-05-31 18:34:25,048 [gpodder.download] ERROR: HTTP 400 while downloading "17 Marr Gan and the Wastes": 400 Client Error: Bad Request for url: https://s3.eu-west-2.amazonaws.com/sanspantsplus/Hey-DnD-Adam/17+Marr+Gan+and+the+Wastes.mp3
Traceback (most recent call last):
  File "C:\Program Files (x86)\gPodder\lib\python3.9\site-packages\gpodder\download.py", line 277, in retrieve_resume
    resp.raise_for_status()
  File "C:/msys64/home/appveyor/_gpodder_build_root/mingw32/lib/python3.9/site-packages/requests/models.py", line 943, in raise_for_status
requests.exceptions.HTTPError: 400 Client Error: Bad Request for url: https://s3.eu-west-2.amazonaws.com/sanspantsplus/Hey-DnD-Adam/17+Marr+Gan+and+the+Wastes.mp3

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Program Files (x86)\gPodder\lib\python3.9\site-packages\gpodder\download.py", line 796, in run
    headers, real_url = downloader.retrieve_resume(self.tempname, self.status_updated)
  File "C:\Program Files (x86)\gPodder\lib\python3.9\site-packages\gpodder\download.py", line 343, in retrieve_resume
    headers, real_url = downloader.retrieve_resume(url,
  File "C:\Program Files (x86)\gPodder\lib\python3.9\site-packages\gpodder\download.py", line 279, in retrieve_resume
    raise gPodderDownloadHTTPError(url, resp.status_code, str(e))
gpodder.download.gPodderDownloadHTTPError: ('https://s3.eu-west-2.amazonaws.com/sanspantsplus/Hey-DnD-Adam/17+Marr+Gan+and+the+Wastes.mp3', 400, '400 Client Error: Bad Request for url: https://s3.eu-west-2.amazonaws.com/sanspantsplus/Hey-DnD-Adam/17+Marr+Gan+and+the+Wastes.mp3')
```

I know the files exist - going to that URL in a browser gives me the file and I can play it etc.
The podcast is password protected, but I don't believe that's the issue.  I'm suspicious about the `+` signs - all the cases that don't work contain `+` in the URL.  A quick skim on the internet implies that `+` itself is allowed in URLs, but maybe it's not be handled properly?

Any thoughts greatly received, thanks!

**Comment by JKAbrams** [comment]
Authentication is set up in `src/gpodder/download.py` on line `247` and used on line `272`.
One simple solution is to just introduce handler for this specific error on line `280` if `auth` is not `None` and try again.

**Comment by auouymous** [comment]
Fixed by #1300.
